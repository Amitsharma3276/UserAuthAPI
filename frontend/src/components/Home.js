import React from "react";
import { useEffect, useState } from "react";
import { homeService } from "../services/homeService";
import HomeData from "./homeData";
import HomeError from "./homeError";
import HomeNoData from "./homeNoData";
import { logoutService } from "../services/logoutService";

const initialState = {
  token: "blank",
  data: [],
  userName: "blank",
  error: false,
};

const Home = () => {
  const [homeInfo, setHomeInfo] = useState(initialState);

  useEffect(() => {
    // const fetchData = async (homeInfo) => {
    //   const response = await homeService(homeInfo);
    //   return response;
    // };
    // const res = fetchData(homeInfo);
    // console.log(res);
    // const responseData = res.text;

    homeService()
      .then(function (response) {
        return response.text();
      })
      .then(function (responseData) {
        if (responseData === "You must login to see this page") {
          setHomeInfo({
            token: "null",
            data: ["sorry", "You must login to see this page"],
            userName: "Stranger",
            error: true,
          });
        } else if (responseData === "Unauthorized") {
          setHomeInfo({
            token: "blank",
            data: ["Hey", "You've Entered Somewhere you shouldn't be"],
            userName: "blank",
            error: true,
          });
        } else {
          const data = responseData.replace("}", "");
          const arr = data.split("#");
          const token = arr[1];
          let tmp = arr[0];
          let userName = arr[2];
          userName = userName.split(":")[1];
          tmp = tmp.replace("{", " ");
          let center = tmp.replace("=", " # Cost : ");
          let centerArr = center.split(",");
          for (let i = 0; i < centerArr.length; i++) {
            centerArr[i] = centerArr[i].replace("=", "# Cost : ");
          }
          console.log(center);
          setHomeInfo({
            token: token,
            data: centerArr,
            userName: userName,
            error: false,
          });
        }
      });
  }, []);

  return (
    <div>
      {homeInfo.error && (
        <HomeError homeData={homeInfo} handleSignOut={logoutService} />
      )}
      {!homeInfo.error && !homeInfo.data && (
        <HomeNoData homeData={homeInfo} handleSignOut={logoutService} />
      )}
      {!homeInfo.error && homeInfo.data && (
        <HomeData homeData={homeInfo} handleSignOut={logoutService} />
      )}
    </div>
  );
};

export default Home;
