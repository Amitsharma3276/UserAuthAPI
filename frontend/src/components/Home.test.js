import React from "react";
import { shallow } from "enzyme";
import Home from "./Home";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { act } from "react-dom/test-utils";
import { render } from "@testing-library/react";
import { unmountComponentAtNode } from "react-dom";

configure({ adapter: new Adapter() });
let container = null;
beforeEach(() => {
  //render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe("<Home /> test suite", () => {
  const wrapper = shallow(<Home />);

  //SNAPSHOT
  it("should match the snapshot", () => {
    act(() => {
      render(wrapper, container);
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
