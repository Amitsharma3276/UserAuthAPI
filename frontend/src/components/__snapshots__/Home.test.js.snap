// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> test suite should match the snapshot 1`] = `"<div><div class=\\"container displayContainer\\" style=\\"height:92vh\\"><center><h1>Welcome blank</h1></center><p align=\\"right\\">blank</p><a href=\\"/login\\"><button class=\\"nav-link\\" to=\\"/login\\">Logout</button></a><br/><center><h2>These are the vaccination centers we found near you</h2></center><h3>Vaccinations Centers : </h3><div></div></div></div>"`;
